app/src/main/java/com/appdev/eateryblueandroid/ui/components/onboarding/OnboardingViewPager.kt

```diff
@@ -0,0 +1,189 @@
+                                offsetY = .35f,
+                                rotate = -24f
+                            ),
+                            IconData(
+                                painter = painterResource(id = R.drawable.ic_item),
+                                offsetX = .6f,
+                                offsetY = .1f,
+                                rotate = 0f
+                            )
+                        )
+                    )
+                }
+                1 -> {
+                    OnboardingPage(
+                        num = 1,
+                        pagerOffset = calculateCurrentOffsetForPage(1),
+                        icons = listOf(
+                            IconData(
+                                painter = painterResource(id = R.drawable.ic_selected_off),
+                                offsetX = -.55f,
+                                offsetY = -.35f,
+                                rotate = -12f
+                            ),
+                            IconData(
+                                painter = painterResource(id = R.drawable.ic_bell),
+                                offsetX = .6f,
+                                offsetY = .28f,
+                                rotate = 0f
+                            )
+                        )
+                    )
+                }
+                2 -> {
+                    OnboardingPage(
+                        num = 2,
+                        pagerOffset = calculateCurrentOffsetForPage(2),
+                        icons = listOf(
+                            IconData(
+                                painter = painterResource(id = R.drawable.ic_clock),
+                                offsetX = -.52f,
+                                offsetY = .2f,
+                                rotate = -12f
+                            ),
+                            IconData(
+                                painter = painterResource(id = R.drawable.ic_watch_big),
+                                offsetX = .55f,
+                                offsetY = -.15f,
+                                rotate = 24f
+                            )
+                        )
+                    )
+                }
+                3 -> {
+                    LoginPage(
+                        profileViewModel = profileViewModel,
+                        pagerOffset = calculateCurrentOffsetForPage(3)
+                    )
+                }
+            }
+        }
+
+        HorizontalPager(
+            count = 4,
+            state = thirdPagerState,
+            userScrollEnabled = false,
+            modifier = Modifier.then(modifier)
+        ) { page ->
+            if (page < 3)
+                OnboardingNextButton(
+                    num = page,
+                    pagerState = secondPagerState,
+                    pagerOffset = calculateCurrentOffsetForPage(page)
+                )
+        }
+    }
```

Comment:
I originally designed the page, header, and button as one composable with one scaffold like you described. It was far cleaner, but it had a bug in which the horizontal pager would fail to render the next page until the current page was snapped into place. As a result, the phone on the side out of view would appear and disappear quite apparently which looked bad. 

It seems horizontalpager doesn't mix well with the graphicsLayer modifier when used to offset images (such as the phone) which is why such a bug was occurring.

Reply:
Since there's a next button as well you can opt for disabling swiping by setting `userScrollEnabled = false`, which may play better with the Animation/can let us use the scaffold